src/iotconnect_discovery.c

```diff
@@ -56,16 +57,19 @@ static bool split_url(IotclDiscoveryResponse *response) {
 }
 
 IotclDiscoveryResponse *iotcl_discovery_parse_discovery_response(const char *response_data) {
+    cJSON *base_url_cjson;
```

Comment:
This change to an uninitialised variable can lead to a bad dereference on line 80, i.e.

char *jsonBaseUrl = base_url_cjson->valuestring;